py3status/module.py

```diff
@@ -479,6 +472,20 @@ def process_composite(self, response):
                 else:
                     item[key] = value
 
+            # remove urgent if not allowed
+            if not self.allow_urgent:
+                if "urgent" in item:
+                    del item["urgent"]
+            # if urgent we want to set this to all parts
+            elif urgent:
+                if self.urgent_background:
+                    item["background"] = self.urgent_background
+                    item["color"] = "#FFFFFF"
```

Comment:
should this really be configurable with `urgent_foreground`? Currently it won't be possible to set `urgent_background` to white for example.

Reply:
let's request an official statement from @ultrabug 😄 In my opinion, py3status core should support every feature that i3 and i3status have, including i3-gaps fork.

Reply:
hmm on the other hand, strictly speaking this is not a feature that i3status supports but py3status doesn't, you are disabling urgent and faking it with custom drawing... This is not like https://github.com/ultrabug/py3status/pull/1486 where border was something that i3-gaps provides as a feature that was impossible to configure with py3status.

I'm not so sure anymore about this PR, let @ultrabug decide.

Reply:
What advantages does the `urgent` carry? 99% of the time, it paints the background.

Other than not using `urgent_background` in container modules (i.e. frame and group), the behavior looks mostly same to me. Don't take my word for it, but I think urgent was made for i3 workspaces. I don't recall i3status have urgent modules.

Reply:
> This is not like #1486 where border was something that i3-gaps provides as a feature that was impossible to configure with py3status.

This is exactly what I don't like too here I'm sorry.

Also I must have looked it up too fast but I can't find the `urgent_background` support in the i3-gaps documentation?

Reply:
i3-gaps does not provide this feature. However, it provides background and border. We replace `urgent` with custom border/background stuff... only if they're user-defined.

Scenario:
>You're sitting in your open room toilet 15 feet away from your new computer... You glance at your monitor from a distance... There, you saw that you have an urgent module on your bar... but you can't tell which it is... because they all look same to you. You continue to poop sadly.

This is where this feature `urgent_background` would come in handy... possibly more handier than a toilet paper in your hands. With this, we can identify urgent modules much faster. A second can make so much difference. Some would even say it made our day.

Reply:
I understand both reasoning, on the one hand it is tempting to diverge as little as possible from the upstream i3, but on the other hand i3-gaps is very popular and it does allow setting custom background and borders, and this opens endless possibilities for customization, including customizing urgent colors. This is a core feature, and it potentially makes py3status much more appealing to users who like to customize everything, so I welcome this PR, but at the same time I can understand if we end up not merging this.

Reply:
@maximbaz I didn't mean to push the second commit. Compare the time. I already did it before you left a foreground comment. At least, you can try it now.

Reply:
After you pointed this out, I think `urgent_color` is more consistent

Reply:
I think new option `foreground` to fully override `color` would be nice. If
you try `color='violet'` in `loadavg`, it does not touch the existing colors.

And `urgent_color` is vague imho as it can be misinterpreted for background.
I like my options.. excluding len() placeholders.. to be bit more explicit.

In the end, I think keeping it as-is with `urgent_foreground` and not
worrying about `foreground` because this part could be in `core.py`...
and it is easier to know what we want (ie background vs foreground... and not
background vs color).